src/middlewares/helper.js

```diff
@@ -48,4 +52,74 @@ const validateInput = (request, response, next) => {
   }
   return next();
 };
-export { sendgrid, verifyPasswordResetToken, validateInput };
+
+const verifyUserStatus = (request, response, next) => {
+  const { role } = request.userData.payload;
+  if (role !== 'superAdmin') {
+    return response.status(403).json({
+      status: 'Fail',
+      message: 'You are not authurized to perform this operation'
+    });
+  }
+  return next();
+};
+
+const validateUpdateInputs = (request, response, next) => {
+  const { email, role } = request.body;
+  if (!email || !email.trim()) {
+    return response.status(400).json({
+      status: 'Fail',
+      message: 'Email field is required'
+    });
+  }
+  if (!role || !role.trim()) {
+    return response.status(400).json({
+      status: 'Fail',
+      message: 'Role field is required'
+    });
+  }
+  return next();
+};
+
+const verifyAdminOrSuperAmin = (request, response, next) => {
+  const { role } = request.userData.payload;
+  if (role === 'superAdmin' || role === 'admin') {
+    return next();
+  }
+  return response.status(403).json({
+    status: 'Fail',
+    message: 'You are not authurized to perform this operation'
+  });
+};
+
+const findOneUser = async (request, response, next) => {
+  const { email } = request.body;
+  try {
+    const result = await User.findOne({
+      where: {
+        email,
+      }
+    });
+    if (!result) {
+      return response.status(404).json({
+        status: 'Fail',
+        message: 'User does not exist'
+      });
+    }
+    return next();
+  } catch (error) {
+    response.status(500).json({
+      status: 'Fail',
+      message: errorResponse(['internal server error, please try again later']),
+    });
+  }
+};
+export {
+  sendgrid,
+  verifyPasswordResetToken,
+  validateInput,
+  verifyUserStatus,
+  validateUpdateInputs,
+  verifyAdminOrSuperAmin,
```

Comment:
Typo => `verifyAdminOrSuperAmin` from above 👆

Reply:
Resolved. Thanks for pointing it out.